datashader/utils.py

```diff
@@ -469,10 +479,127 @@ def mesh(vertices, simplices):
-
+    assert len(vertices.columns) > 2 or simplices.values.shape[1] > 3, \
+        'If no vertex weight column is provided, a triangle weight column is required.'
 
-    if isinstance(vertices, dd.DataFrame) and isinstance(simplices, dd.DataFrame):
+    if has_dask and isinstance(vertices, DaskDataFrame) and isinstance(simplices, DaskDataFrame):
         return _dd_mesh(vertices, simplices)
 
     return _pd_mesh(vertices, simplices)
+
+
+def pyspark_type_to_pandas_type(data_type):
+    """
+    Map a PySpark type to a pandas (numpy) type
+    """
+    import pyspark.sql.types as T
+    if isinstance(data_type, T.ArrayType):
+        return np.object
+    if isinstance(data_type, T.BooleanType):
+        return np.bool
+    if isinstance(data_type, T.ByteType):
+        return np.int8
+    if isinstance(data_type, T.DecimalType):
+        return np.float64
+    if isinstance(data_type, T.DoubleType):
+        return np.float64
+    if isinstance(data_type, T.FloatType):
+        return np.float32
+    if isinstance(data_type, T.IntegerType):
+        return np.int32
+    if isinstance(data_type, T.LongType):
+        return np.int64
+    if isinstance(data_type, T.MapType):
+        return np.object
+    if isinstance(data_type, T.NullType):
+        return np.object
+    if isinstance(data_type, T.ShortType):
+        return np.int16
+    if isinstance(data_type, T.StructType):
+        return np.object
+    if isinstance(data_type, T.StringType):
+        return np.object
+    if isinstance(data_type, T.TimestampType):
+        return np.datetime64
+    raise TypeError("Unrecognized PySpark type %r" % data_type)
+
+
+def nullsafe_pandas_type(numpy_type):
+    """
+    Map a pandas (numpy) type to its nullsafe equivalent
+    """
+    # objects and floating point naturally support null values
+    if numpy_type in (np.object, np.float32, np.float64):
+        return numpy_type
+
+    # all other types can be safely captured by object **when used in pandas dataframes**
+    return np.object
+
+
+def pyspark_to_pandas_schema(schema):
+    """
+    Map a pyspark schema to a pandas schema of {column name: type} and {column name: nullsafe type}
+    """
+    pandas_schema, nullsafe_schema = dict(), dict()
+    for field in schema:
+        pandas_type = pyspark_type_to_pandas_type(field.dataType)
+        pandas_schema[field.name] = pandas_type
+        nullsafe_schema[field.name] = nullsafe_pandas_type(pandas_type) if field.nullable \
+            else pandas_type
+    return pandas_schema, nullsafe_schema
+
+
+def empty_typed_dataframe(schema):
+    """
+    Create an empty, typed dataframe from a pandas schema
+    """
+    return pd.DataFrame({name: np.empty(0, dtype=dtype) for name, dtype in schema.items()})
+
+
+def serialized_rows_to_pandas(pandas_schema, nullsafe_schema, rows):
+    """
+    Read a serialized collection of rows (an iterator of pyspark.sql.Row objects) as a pandas
+    dataframe
+    """
```

Comment:
The docstring says what it does, but not why anyone would want or need this.  Plus the name should clarify that it's about spark only; it sounds like it's about the Pandas backend.